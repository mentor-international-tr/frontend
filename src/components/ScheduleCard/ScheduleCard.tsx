import React from "react";
import {
  MIScheduleCard,
  MICardContent,
  Header,
  Title,
  ActionButtonsWrapper,
  MessageBox,
  DateBox,
  Date,
} from "./ScheduleCardStyles";
import {
  Card,
  CardContent,
  IconButton,
} from "@material-ui/core";
export interface ScheduleCardProps {
  title: string;
  message: string;
  dates: string[];
}

const ScheduleCard: React.SFC<ScheduleCardProps> = ({
  title,
  message,
  dates,
}) => {
  return (
    <MIScheduleCard>
      <MICardContent>
        <Header>
          <Title>{title}</Title>
          <ActionButtonsWrapper>
            <IconButton aria-label="Edit">
              <i className="fas fa-pencil-alt" />
            </IconButton>
            <IconButton aria-label="Delete">
              <i
                className="fa fa-trash"
                aria-hidden="true"
              />
            </IconButton>
          </ActionButtonsWrapper>
        </Header>
        <MessageBox>{message}</MessageBox>
        <DateBox>
          {dates.map((date: string) => {
            return <Date>{date}</Date>;
          })}
        </DateBox>
      </MICardContent>
    </MIScheduleCard>
  );
};

export default ScheduleCard;
