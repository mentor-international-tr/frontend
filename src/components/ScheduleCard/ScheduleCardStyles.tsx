import styled from "styled-components";
import vars from "lib/styles/variables";
import { Card, CardContent } from "@material-ui/core";

const MIScheduleCard = styled(Card)`
  display: flex;
  flex-direction: column;
  align-items: center;
  max-width: 350px;
  background: ${vars.white};
`;

const MICardContent = styled(CardContent)`
  background: ${vars.white};
  width: 100%;
`;

const Header = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const Title = styled.h2``;

const ActionButtonsWrapper = styled.div``;

const MessageBox = styled.div`
  border: 1px solid ${vars.grey};
  border-radius: 2px;
  padding: ${vars.space_md} ${vars.space_xl}
    ${vars.space_md} ${vars.space_xl};
  position: relative;
  margin-bottom: ${vars.space_md};
  &:before {
    content: "Message";
    position: absolute;
    background: ${vars.white};
    top: -12px;
  }
`;

const DateBox = styled.div`
  border: 1px solid ${vars.grey};
  border-radius: 2px;
  padding: ${vars.space_sm};
  position: relative;
  &:before {
    content: "Dates to Send";
    position: absolute;
    background: ${vars.white};
    top: -12px;
  }
`;
const Date = styled.p`
  margin: 0;
`;

export {
  MIScheduleCard,
  MICardContent,
  Header,
  Title,
  ActionButtonsWrapper,
  MessageBox,
  DateBox,
  Date,
};
