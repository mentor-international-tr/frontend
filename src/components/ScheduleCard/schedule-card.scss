@import "src/lib/styles/variables.scss";
.schedule-card {
    display: flex;
    flex-direction: column;
    align-items: center;
    max-width: 350px;
    background: $white;
}

.schedule-card__content {
    background: $white;
    width: 100%;
}
.schedule-card__header {
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.schedule-card__msg-box {
    border: 1px solid $grey;
    border-radius: 2px;
    padding: $space-md $space-md $space-xl;
    position: relative;
    margin-bottom: $space-md;
    &:before {
        content: "Message";
        position: absolute;
        background: $white;
        top: -12px;
    }
}
.schedule-card__date-box {
    border: 1px solid $grey;
    border-radius: 2px;
    padding: $space-sm;
    position: relative;
    &:before {
        content: "Dates to Send";
        position: absolute;
        background: $white;
        top: -12px;
    }
}
.schedule-card__date {
    margin: 0;
}