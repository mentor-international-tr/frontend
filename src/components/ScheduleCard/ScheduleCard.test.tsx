import React from "react";
import { render, cleanup } from "react-testing-library";
import ScheduleCard from "./ScheduleCard";
import "jest-dom/extend-expect";

afterEach(cleanup);

describe("testing <ScheduleCard /> component", () => {
  it("creates snapshot", () => {
    const { asFragment } = render(
      <ScheduleCard
        title="Class Reminder"
        message={
            "Don't forget, class tonight at 8 pm! Don't forget, class tonight at 8 pm!"
        }
        dates={["July 20th", "Nov 24th"]}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
