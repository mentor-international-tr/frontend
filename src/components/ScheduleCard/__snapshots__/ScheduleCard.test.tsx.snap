// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <ScheduleCard /> component creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1 sc-bdVaJa gTgABg"
  >
    <div
      class="MuiCardContent-root-29 sc-bwzfXH jngNH"
    >
      <header
        class="sc-htpNat fAaqcp"
      >
        <h2
          class="sc-bxivhb bUZGRr"
        >
          Class Reminder
        </h2>
        <div
          class="sc-ifAKCX cFlEyZ"
        >
          <button
            aria-label="Edit"
            class="MuiButtonBase-root-36 MuiIconButton-root-30"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label-35"
            >
              <i
                class="fas fa-pencil-alt"
              />
            </span>
            <span
              class="MuiTouchRipple-root-39"
            />
          </button>
          <button
            aria-label="Delete"
            class="MuiButtonBase-root-36 MuiIconButton-root-30"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label-35"
            >
              <i
                class="fa fa-trash"
              />
            </span>
            <span
              class="MuiTouchRipple-root-39"
            />
          </button>
        </div>
      </header>
      <div
        class="sc-EHOje iYyHZw"
      >
        Don't forget, class tonight at 8 pm! Don't forget, class tonight at 8 pm!
      </div>
      <div
        class="sc-bZQynM bUXyNz"
      >
        <p
          class="sc-gzVnrw deJinG"
        >
          July 20th
        </p>
        <p
          class="sc-gzVnrw deJinG"
        >
          Nov 24th
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
